salt/client/ssh/__init__.py

```diff
@@ -1175,7 +1175,12 @@ def _cmd_str(self):
         '''
         Prepare the command string
         '''
-        sudo = 'sudo' if self.target['sudo'] else ''
+        if self.target['sudo']:
+            sudo = 'sudo'
+            # set the sudo promtp parameter with the unique prompt if the password is set
+            sudo += ' -p \"{0}\"'.format(shell.SUDO_PROMPT) if self.target['passwd'] else ''
```

Comment:
What's wrong avoid `+=` and do just `sudo = ' sudo -p \"{......` ? That said:

```python
if self.target.get('sudo'):
    sudo = self.target.get('passwd') and 'sudo -p \"{0}\"'.format(shell.SUDO_PROMPT) or ''
else:
    sudo = ''
```

Reply:
Is there a reason you use self.target.get('sudo') in your exmple in stead of self.target['sudo']?

Reply:
The code above I've made has less nested ifs and in case `sudo` key is not in `target` for whatever reason, it won't crash but will return `None`. The rest is the same as yours. Just my bad that I forgot to write `'sudo'` at the end and wrote `''`. Fixed.

Reply:
I had to read your example a couple of times before I understood how it works. Are you sure you want a 

```python
x = test and a or b
```

in stead of 

```python
x = a if test else b
```
because it does not really help readability?
Your "and + or" construction functions as an "if else" only you depend on the execution order. In most code standards this is not allowed.